src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java

```diff
@@ -30,21 +33,39 @@
  *
  * @author <a href="mailto:simon@redhillconsulting.com.au">Simon Harris</a>
  * @author Oliver Burn
+ * @author Vikramaditya Kukreja
  */
 public final class PackageDeclarationCheck extends AbstractCheck {
 
     /**
      * A key is pointing to the warning message text in "messages.properties"
      * file.
      */
-    public static final String MSG_KEY = "missing.package.declaration";
+    public static final String MSG_KEY_MISSING = "missing.package.declaration";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY_MISMATCH = "mismatch.package.directory";
 
     /** Line number used to log violation when no AST nodes are present in file. */
     private static final int DEFAULT_LINE_NUMBER = 1;
 
     /** Is package defined. */
     private boolean defined;
 
+    /** Whether to ignore the directory name matching package name of source file. */
+    private boolean ignoreMismatch = true;
```

Comment:
If I do go ahead with `matchDirectoryStructure`, then in the setter method would be like this - ? 
`setMatchDirectoryStructure`. Would this be correct for Javabeans Reflection property?